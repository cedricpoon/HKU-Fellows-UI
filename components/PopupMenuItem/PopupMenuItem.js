import React, { PureComponent } from 'react';
import { Platform, StyleSheet, TouchableNativeFeedback } from 'react-native';
import { Button } from 'native-base';
import variable from "hkufui/native-base-theme/variables/platform";

const styles = StyleSheet.create({
  button: {
    width: Platform.OS === 'android' ? '100%' : null
  }
});

class PopupMenuItem extends PureComponent {
  render() {
    return (
      <Button
        style={styles.button}
        background={TouchableNativeFeedback.Ripple(variable.androidRippleColor, false)}
        {...this.props}
      />
    );
  }
}

export default PopupMenuItem;
