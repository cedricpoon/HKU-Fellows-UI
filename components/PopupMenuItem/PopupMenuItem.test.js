import React from 'react';
import { shallow } from 'enzyme';

import PopupMenuItem from './PopupMenuItem';

describe('Testing PopupMenuItem Component', () => {
  it('renders', () => {
    const wrapper = shallow(
      <PopupMenuItem />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
