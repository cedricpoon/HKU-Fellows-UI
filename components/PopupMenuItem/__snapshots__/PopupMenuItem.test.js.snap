// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PopupMenuItem Component renders 1`] = `
<Styled(Button)
  background={Object {}}
  style={
    Object {
      "width": null,
    }
  }
/>
`;
